adal/src/main/java/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -630,23 +521,47 @@ public void acquireToken(String resource, String clientId, @Nullable String redi
                              @Nullable final String loginHint, @Nullable final PromptBehavior prompt, @Nullable String extraQueryParameters,
                              @Nullable final String claims, final AuthenticationCallback<AuthenticationResult> callback) {
+
+        acquireToken(resource, clientId, redirectUri, loginHint, prompt, extraQueryParameters,
+                claims, callback, EventStrings.ACQUIRE_TOKEN_10, null);
+
+    }
+
+    private void acquireToken(final String resource, final String clientId, @Nullable String redirectUri,
+                              @Nullable final String loginHint, @Nullable final PromptBehavior prompt,
+                              @Nullable String extraQueryParameters, @Nullable final String claims,
+                              final AuthenticationCallback<AuthenticationResult> callback, String apiEventString,
+                              final IWindowComponent fragment){
+
         throwIfClaimsInBothExtraQpAndClaimsParameter(claims, extraQueryParameters);
 
-        if (checkPreRequirements(resource, clientId, callback)
-                && checkADFSValidationRequirements(loginHint, callback)) {
+        if (checkPreRequirements(resource, clientId, callback) && checkADFSValidationRequirements(loginHint, callback)) {
             redirectUri = getRedirectUri(redirectUri);
             final String requestId = Telemetry.registerNewRequest();
-            final APIEvent apiEvent = createApiEvent(mContext, clientId, requestId, EventStrings.ACQUIRE_TOKEN_10);
+            final APIEvent apiEvent = createApiEvent(mContext, clientId, requestId, apiEventString);
             apiEvent.setPromptBehavior(prompt);
-            apiEvent.setLoginHint(loginHint);
+
+            String mergedClaims = null;
+            try {
+                mergedClaims = mergeClaimsWithClientCapabilities(claims);
+            } catch (JSONException e) {
+                callback.onError(new AuthenticationException(ADALError.JSON_PARSE_ERROR, e.getMessage(), e));
+            }
 
             final AuthenticationRequest request = new AuthenticationRequest(mAuthority, resource,
                     clientId, redirectUri, loginHint, prompt, extraQueryParameters,
-                    getRequestCorrelationId(), getExtendedLifetimeEnabled(), claims);
-            request.setUserIdentifierType(UserIdentifierType.LoginHint);
+                    getRequestCorrelationId(), getExtendedLifetimeEnabled(), mergedClaims);
             request.setTelemetryRequestId(requestId);
-            createAcquireTokenRequest(apiEvent).acquireToken(null, false, request, callback);
+            setAppInfoToRequest(request);
+
+            if(!TextUtils.isEmpty(loginHint)) {
+                apiEvent.setLoginHint(loginHint);
+                request.setUserIdentifierType(UserIdentifierType.LoginHint);
+            }
+
+            createAcquireTokenRequest(apiEvent).acquireToken(fragment, false, request, callback);
```

Comment:
second param here is useDialog, some api's like ACQUIRE_TOKEN_7 pass true to it

Reply:
Thanks @addev-ashish for catching this. Updated to add this param as well